crates/net/network/src/transactions/mod.rs

```diff
@@ -991,12 +987,19 @@ where
                 let _ = response.send(Ok(PooledTransactions::default()));
                 return
             }
-            let transactions = self.pool.get_pooled_transactions_as::<N::PooledTransaction>(
-                request.0,
-                GetPooledTransactionLimit::ResponseSizeSoftLimit(
-                    self.transaction_fetcher.info.soft_limit_byte_size_pooled_transactions_response,
-                ),
-            );
+            let transactions = self
+                .pool
+                .get_pooled_transaction_elements(
+                    request.0,
+                    GetPooledTransactionLimit::ResponseSizeSoftLimit(
+                        self.transaction_fetcher
+                            .info
+                            .soft_limit_byte_size_pooled_transactions_response,
+                    ),
+                )
+                .into_iter()
+                .map(|tx| tx.into_signed())
+                .collect::<Vec<_>>();
```

Comment:
this collect is a bit redundant and `get_pooled_transaction_elements` was a workaround to avoid the collect here